test_runner/fixtures/pg_stats.py

```diff
@@ -1,21 +1,30 @@
-from typing import List
+from typing import List, Tuple
 
 import pytest
 
 
 class PgStatTable:
+    __slots__: Tuple[str, ...] = (
+        "table",
+        "columns",
+        "additional_query",
+        "_query",
+    )
```

Comment:
a bit

Reply:
Ok, but is it worth it?

Reply:
https://faun.pub/improve-python-class-performance-using-slots-d3e986a282cf

I can share this article as a reference. It requires "one-line", but adds a perf boost

Reply:
I guess that we spend much more effort discussing these lines. `__slots__` adds some magic to these classes. On the one hand, it improves the performance of instating new objects, on the other hand, you have to update the slots while adding new properties into the primary class and its descendants.

Reply:
> I guess that we spend much more effort discussing these lines.

And you'd be right if our core IP was this python code. Then it'd make a lot of sense to try to squeeze every last bit of performance from it. But the reality is very different. We use python for tests because it's simple, forgiving and has decent testing capabilities. Some of us might not be aware of features like `__slots__` and other rather specific band-aids it needs to stay afloat when you're trying to develop a backend.

Thus I'd say that simplicity is more important, especially if there's no strong evidence (measurements, graphs, etc) that this change helps a lot.